addons/account/models/account_move.py

```diff
@@ -2287,17 +2288,26 @@ def _stolen_move(self, vals):
             if command[0] == Command.SET:
                 yield from self.env['account.move.line'].browse(command[2]).move_id.ids
 
+    def _get_protected_vals(self, vals, records):
+        protected = set()
+        for fname in vals:
+            field = records._fields.get(fname)
+            if field.inverse or (field.compute and not field.readonly):
+                protected.update(self.pool.field_computed.get(field, [field]))
+        return [(protected, rec) for rec in records] if protected else []
+
     @api.model_create_multi
     def create(self, vals_list):
         if any('state' in vals and vals.get('state') == 'posted' for vals in vals_list):
             raise UserError(_('You cannot create a move already in the posted state. Please create a draft move and post it after.'))
         container = {'records': self}
         with self._check_balanced(container):
-            with self._sync_dynamic_lines(container):
+            with self.env.protector() as protector, self._sync_dynamic_lines(container):
                 for vals in vals_list:
                     self._sanitize_vals(vals)
                 stolen_moves = self.browse(set(move for vals in vals_list for move in self._stolen_move(vals)))
                 moves = super().create(vals_list)
+                protector.protect([protected for vals, move in zip(vals_list, moves) for protected in self._get_protected_vals(vals, move)])
```

Comment:
```suggestion
            with ExitStack() as stack:
                stack.enter(self._sync_dynamic_lines(container))
                for vals in vals_list:
                    self._sanitize_vals(vals)
                stolen_moves = self.browse(set(move for vals in vals_list for move in self._stolen_move(vals)))
                moves = super().create(vals_list)
                stack.enter(self.env.protecting([protected for vals, move in zip(vals_list, moves) for protected in self._get_protected_vals(vals, move)])
```

Reply:
Is this correct? This is what I believe it will do:
* enter `_sync_dynamic_lines`
* call `super().create`
* enter `protecting`
* exit `protecting`
* exit `_sync_dynamic_lines`: this is the step that needs the protection

Reply:
But that doesn't change anything?

Reply:
Edited. Now it exits protecting after exiting sync dynamic lines.

Reply:
Thank you, it is much easier indeed :+1: